--- a/source/items/projectiles.c
+++ b/source/items/projectiles.c
@@ -561,7 +561,6 @@
 	struct tag_block const *attachments;
 	real detonation_ticks;
 	real arming_ticks;
-	real initial_velocity;
 	short attachment_index;
 	boolean underwater;
 
@@ -614,16 +613,11 @@
 		}
 	}
 
-	initial_velocity = definition->projectile.initial_velocity;
-	projectile->object.translational_velocity.i =
-		projectile->object.forward.i * initial_velocity +
-		projectile->object.translational_velocity.i;
-	projectile->object.translational_velocity.j =
-		projectile->object.forward.j * initial_velocity +
-		projectile->object.translational_velocity.j;
-	projectile->object.translational_velocity.k =
-		projectile->object.forward.k * initial_velocity +
-		projectile->object.translational_velocity.k;
+	point_from_line3d(
+		(real_point3d *)&projectile->object.translational_velocity,
+		&projectile->object.forward,
+		definition->projectile.initial_velocity,
+		(real_point3d *)&projectile->object.translational_velocity);
 
 	underwater = scenario_location_underwater(
 		&projectile->object.location,
