diff --git a/source/cache/xbox_texture_cache.c b/source/cache/xbox_texture_cache.c
index 58eef577..672a3ce5 100644
--- a/source/cache/xbox_texture_cache.c
+++ b/source/cache/xbox_texture_cache.c
@@ -208,7 +208,6 @@ struct xbox_texture_cache_texture
 	short read_request_handle;
 	boolean loaded;
 	boolean used;
-	byte reserved006[2];
 	struct bitmap_data *bitmap;
 	D3DBaseTexture hardware_format;
 };
