diff --git a/source/rasterizer/rasterizer_lights.c b/source/rasterizer/rasterizer_lights.c
index e13b301a..61985fb8 100644
--- a/source/rasterizer/rasterizer_lights.c
+++ b/source/rasterizer/rasterizer_lights.c
@@ -904,7 +904,7 @@ void rasterizer_lens_flares_draw(
 					real_vector3d mirror;
 					real depth;
 					real occlusion_fraction;
-					real light_brightness;
+					real light_brightness = 1.0f;
 					real corona_rotation;
 					real screen_rotation;
 					real cosine_scale;
@@ -925,18 +925,13 @@ void rasterizer_lens_flares_draw(
 
 					occlusion_fraction = *occlusion_test_result*(1.0f/255.0f);
 
-					if (definition->far_fade_distance > 0.0f)
-					{
-						light_brightness = PIN(
+					light_brightness = definition->far_fade_distance > 0.0f ?
+						PIN(
 							(depth-definition->far_fade_distance)/
 								(definition->near_fade_distance-definition->far_fade_distance),
 							0.0f,
-							1.0f);
-					}
-					else
-					{
-						light_brightness = 1.0f;
-					}
+							1.0f) :
+						1.0f;
 
 					light_brightness *= occlusion_fraction;
 					light_brightness *= uncompress_int8_to_real(
@@ -979,8 +974,9 @@ void rasterizer_lens_flares_draw(
 								reflection_index,
 								struct lens_flare_reflection);
 							real brightness_lower_bound = reflection->brightness_lower_bounds;
+							real brightness_upper_bound = reflection->brightness_upper_bounds;
 							real brightness = (brightness_lower_bound+
-								(reflection->brightness_upper_bounds-brightness_lower_bound)*light_scale)*
+								(brightness_upper_bound-brightness_lower_bound)*light_scale)*
 								scale_functions[reflection->brightness_scale_function]*light_brightness;
 
 							if (reflection_index == 0)
@@ -991,8 +987,9 @@ void rasterizer_lens_flares_draw(
 							if (brightness > 0.0f)
 							{
 								real radius_lower_bound = reflection->radius_lower_bounds;
+								real radius_upper_bound = reflection->radius_upper_bounds;
 								real radius = radius_lower_bound+
-									(reflection->radius_upper_bounds-radius_lower_bound)*light_scale;
+									(radius_upper_bound-radius_lower_bound)*light_scale;
 								real_argb_color color;
 								real_vector2d radius_scale;
 								real rotation;
@@ -1094,17 +1091,16 @@ void rasterizer_lens_flares_draw(
 								}
 
 								{
-									real offset = reflection->offset;
 									real_point3d point;
+									boolean set_texture_result;
 
-									point.x = offset*mirror.i + position.x;
-									point.y = offset*mirror.j + position.y;
-									point.z = offset*mirror.k + position.z;
+									point_from_line3d(&position, &mirror, reflection->offset, &point);
 
-									if (rasterizer_widget_set_texture(
+									set_texture_result = rasterizer_widget_set_texture(
 										0,
 										definition->primary_map.index,
-										reflection->bitmap_index))
+										reflection->bitmap_index);
+									if (set_texture_result)
 									{
 										break;
 									}
@@ -1138,10 +1134,12 @@ void rasterizer_lens_flares_draw(
 
 		if (rasterizer_debug_options.ray_of_buddha)
 		{
-			for (lens_flare_index = 0; lens_flare_index < local_lens_flare_count; lens_flare_index++)
+			short sun_lens_flare_index;
+
+			for (sun_lens_flare_index = 0; sun_lens_flare_index < local_lens_flare_count; sun_lens_flare_index++)
 			{
 				struct rasterizer_lens_flare_submit_parameters *lens_flare_parameters =
-					lens_flare_submit_parameter_get(lens_flare_index);
+					lens_flare_submit_parameter_get(sun_lens_flare_index);
 
 				if (lens_flare_parameters->internal__occlusion_pixels > 0 &&
 					(lens_flare_parameters->compressed_window_index & _lens_flare_window_index_mask) ==
