--- a/source/effects/decals.c
+++ b/source/effects/decals.c
@@ -212,23 +212,15 @@
 
 /* ---------- headers */
 
-#define projection_sign_from_vector3d projection_sign_from_vector3d_inline
 #define project_point2d project_point2d_inline
-#define triple_product3d triple_product3d_inline
 #define plane2d_from_points plane2d_from_points_inline
-#define plane3d_from_point_and_normal plane3d_from_point_and_normal_inline
-#define plane3d_negate plane3d_negate_inline
 #define REAL_MATH_EXTERNAL_PROJECT_POINT3D
 #include "effects/decals.h"
 #include "cseries/cseries.h"
 #include "math/real_math.h"
 #include "physics/collision_bsp_definitions.h"
-#undef projection_sign_from_vector3d
 #undef project_point2d
-#undef triple_product3d
 #undef plane2d_from_points
-#undef plane3d_from_point_and_normal
-#undef plane3d_negate
 #undef REAL_MATH_EXTERNAL_PROJECT_POINT3D
 
 #include "cseries/errors.h"
@@ -820,16 +812,6 @@
 	return;
 }
 
-boolean projection_sign_from_vector3d(
-	real_vector3d const *n,
-	short projection)
-{
-	match_assert("..\\math\\real_math.h", 848,
-		projection>=_x && projection<=_z);
-
-	return n->n[projection] > 0.0f;
-}
-
 real_point3d *project_point2d(
 	real_point2d const *p2d,
 	real_plane3d const *plane,
@@ -852,16 +834,6 @@
 	return p3d;
 }
 
-real triple_product3d(
-	real_vector3d const *a,
-	real_vector3d const *b,
-	real_vector3d const *n)
-{
-	real_vector3d cross;
-
-	return dot_product3d(cross_product3d(a, b, &cross), n);
-}
-
 real_plane2d *plane2d_from_points(
 	real_plane2d *plane,
 	real_point2d const *point0,
@@ -881,29 +853,6 @@
 
 	return plane;
 }
-
-real_plane3d *plane3d_from_point_and_normal(
-	real_plane3d *plane,
-	real_point3d const *point,
-	real_vector3d const *normal)
-{
-	plane->n = *normal;
-	plane->d = dot_product3d((real_vector3d *)point, &plane->n);
-
-	return plane;
-}
-
-real_plane3d *plane3d_negate(
-	real_plane3d const *p1,
-	real_plane3d *plane)
-{
-	plane->n.i = -p1->n.i;
-	plane->n.j = -p1->n.j;
-	plane->n.k = -p1->n.k;
-	plane->d = -p1->d;
-
-	return plane;
-}
 
 pixel32 real_a_rgb_color_to_pixel32(
 	real alpha,
