--- a/source/bink/bink_playback.c	2026-09-23 19:16:33.494537200 -0700
+++ b/source/bink/bink_playback.c	2026-09-23 19:16:33.518831000 -0700
@@ -1025,8 +1025,11 @@
 
 		if (gamepad)
 		{
-			if (gamepad->buttons[_gamepad_binary_button_start]==1 ||
-				gamepad->buttons[_gamepad_binary_button_back]==1)
+			if (gamepad->buttons[_gamepad_binary_button_start]==1)
+			{
+				button_pressed= TRUE;
+			}
+			else if (gamepad->buttons[_gamepad_binary_button_back]==1)
 			{
 				button_pressed= TRUE;
 			}
