--- a/source/models/models.h
+++ b/source/models/models.h
@@ -15,7 +15,6 @@
 /* ---------- structures */
 
 struct model;
-struct model_geometry_part;
 struct object_marker;
 struct real_matrix4x3;
 struct real_orientation;
@@ -70,8 +69,6 @@
 	short maximum_marker_count);
 void model_build_tangent_matrices(
 	struct model *model);
-void model_geometry_part_build_tangent_matrices(
-	struct model_geometry_part *part);
 void render_model(
 	long model_index,
 	real level_of_detail_pixels,
