diff --git a/source/game/player_control.h b/source/game/player_control.h
--- a/source/game/player_control.h
+++ b/source/game/player_control.h
@@ -152,6 +152,8 @@ void player_control_get_unit_camera_info(
 	struct player_control_unit_camera_info *camera_info);
 long player_control_get_unit_index(
 	short local_player_index);
+short unit_get_local_player_index(
+	long unit_index);
 long player_control_get_desired_weapon(
 	short local_player_index,
 	long unit_index);
diff --git a/source/interface/hud_messaging.h b/source/interface/hud_messaging.h
--- a/source/interface/hud_messaging.h
+++ b/source/interface/hud_messaging.h
@@ -15,6 +15,7 @@ header included in hcex build.
 /* ---------- structures */
 
 struct icon_hud_element_definition;
+union real_argb_color;
 
 /* ---------- prototypes/HUD_MESSAGING.C */
 
@@ -78,6 +79,10 @@ void hud_set_state_text(
 	wchar_t const *message);
 wchar_t *hud_messaging_get_objective(
 	void);
+long hud_get_font_index(
+	void);
+union real_argb_color *hud_get_text_color(
+	union real_argb_color *result);
 
 /* ---------- globals */
 
diff --git a/source/sound/sound_classes.h b/source/sound/sound_classes.h
--- a/source/sound/sound_classes.h
+++ b/source/sound/sound_classes.h
@@ -64,6 +64,10 @@ struct sound_class_definition *sound_class_get(
 
 real sound_class_get_gain(
 	short class_index);
+void sound_class_set_gain(
+	char const *name,
+	real gain,
+	short interpolation_ticks);
 
 void sound_classes_dispose_from_old_map(
 	void);
