diff --git a/source/math/periodic_functions.c b/source/math/periodic_functions.c
--- a/source/math/periodic_functions.c
+++ b/source/math/periodic_functions.c
@@ -321,7 +321,7 @@
 	for (index = 0; index < PERIODIC_FUNCTION_TABLE_SIZE; index++)
 	{
 		x_table[index] = sum;
-		sum += (real_random()+1.0f)*0.25f +
+		sum += real_random()*0.25f + 0.25f +
 			((real)cos(8.2f*_pi*index/PERIODIC_FUNCTION_TABLE_SIZE)+1.0f)*real_random() +
 			((real)cos(10.2f*_pi*index/PERIODIC_FUNCTION_TABLE_SIZE)+1.0f)*real_random() +
 			((real)cos(14.6f*_pi*index/PERIODIC_FUNCTION_TABLE_SIZE)+1.0f)*real_random();
