diff --git a/config/symbols.json b/config/symbols.json
index 885b3b3d..43849da4 100644
--- a/config/symbols.json
+++ b/config/symbols.json
@@ -3607,7 +3607,7 @@
 { "file_offset": 856288, "flags": 32, "name": "_IDirect3DDevice8_SetTransform@12" },
 { "file_offset": 856304, "flags": 32, "name": "_IDirect3DDevice8_GetTransform@12" },
 { "file_offset": 856320, "flags": 32, "name": "_tgaLoad" },
-{ "file_offset": 856400, "flags": 32, "name": "_progress_bar_screen_initialized" },
+{ "file_offset": 856400, "flags": 32, "name": "_progress_bar_is_stuff_ready" },
 { "file_offset": 856416, "flags": 32, "name": "_IDirect3DDevice8_SetRenderState@12" },
 { "file_offset": 856960, "flags": 32, "name": "_IDirect3DDevice8_SetTexture@12" },
 { "file_offset": 856976, "flags": 32, "name": "_IDirect3DDevice8_SetTextureStageState@16" },
@@ -3628,26 +3628,26 @@
 { "file_offset": 857840, "flags": 32, "name": "_IDirect3DDevice8_End@4" },
 { "file_offset": 857856, "flags": 32, "name": "_gen_cloud_coord" },
 { "file_offset": 857904, "flags": 32, "name": "_gen_mask_coord" },
-{ "file_offset": 857968, "flags": 32, "name": "_draw_gravy_layer", "static": true },
-{ "file_offset": 858320, "flags": 32, "name": "_draw_gravy_screen_layer" },
-{ "file_offset": 858416, "flags": 32, "name": "_draw_gravy_base_layer", "static": true },
-{ "file_offset": 858512, "flags": 32, "name": "_progress_bar_alias_surface_texture", "static": true },
-{ "file_offset": 858560, "flags": 32, "name": "_set_blur_texture_coordinates" },
-{ "file_offset": 858720, "flags": 32, "name": "_draw_fullscreen_blur" },
+{ "file_offset": 857968, "flags": 32, "name": "_draw_layer_int", "static": true },
+{ "file_offset": 858320, "flags": 32, "name": "_draw_layer" },
+{ "file_offset": 858416, "flags": 32, "name": "_draw_initial_layer", "static": true },
+{ "file_offset": 858512, "flags": 32, "name": "_this_is_awful", "static": true },
+{ "file_offset": 858560, "flags": 32, "name": "_do_convoluation_coords" },
+{ "file_offset": 858720, "flags": 32, "name": "_draw_fade_layer" },
 { "file_offset": 859024, "flags": 32, "name": "_progress_bar_enable" },
 { "file_offset": 859040, "flags": 32, "name": "_D3DTexture_UnlockRect@8" },
-{ "file_offset": 859056, "flags": 32, "name": "_IDirect3DResource8_Release@4" },
+{ "file_offset": 859056, "flags": 32, "name": "_IDirect3DTexture8_Release@4" },
 { "file_offset": 859072, "flags": 32, "name": "_IDirect3DTexture8_GetLevelDesc@12" },
 { "file_offset": 859088, "flags": 32, "name": "_IDirect3DTexture8_LockRect@20" },
 { "file_offset": 859120, "flags": 32, "name": "_IDirect3DTexture8_UnlockRect@8" },
 { "file_offset": 859136, "flags": 32, "name": "_D3DXMatrixIdentity" },
 { "file_offset": 859200, "flags": 32, "name": "_SetRenderStateSmart" },
 { "file_offset": 859792, "flags": 32, "name": "_SetTextureStageStateSmart" },
-{ "file_offset": 859904, "flags": 32, "name": "_progress_bar_draw", "static": true },
+{ "file_offset": 859904, "flags": 32, "name": "_progress_bar_render", "static": true },
 { "file_offset": 861104, "flags": 32, "name": "_progress_bar_load_loading_texture", "static": true },
-{ "file_offset": 861280, "flags": 32, "name": "_progress_bar_build_glow_texture", "static": true },
+{ "file_offset": 861280, "flags": 32, "name": "_progress_bar_create_mask_texture", "static": true },
 { "file_offset": 861488, "flags": 32, "name": "_progress_bar_eachframe" },
-{ "file_offset": 861680, "flags": 32, "name": "_progress_bar_setup", "static": true },
+{ "file_offset": 861680, "flags": 32, "name": "_progress_bar_make_stuff_ready", "static": true },
 { "file_offset": 862848, "flags": 32, "name": "_progress_bar_display" },
 { "file_offset": 863008, "flags": 32, "name": "_terminal_initialize" },
 { "file_offset": 863072, "flags": 32, "name": "_terminal_dispose" },
diff --git a/source/interface/progress_bar.c b/source/interface/progress_bar.c
index af2cc24c..af8a46ad 100644
--- a/source/interface/progress_bar.c
+++ b/source/interface/progress_bar.c
@@ -27,7 +27,7 @@ symbols in this file:
 000D1100 0050:
 	_tgaLoad (0000)
 000D1150 0010:
-	_code_000d1150 (0000)
+	_progress_bar_is_stuff_ready (0000)
 000D1160 0220:
 	_code_000d1160 (0000)
 000D1380 0010:
@@ -69,23 +69,23 @@ symbols in this file:
 000D1730 0040:
 	_gen_mask_coord (0000)
 000D1770 0160:
-	_code_000d1770 (0000)
+	_draw_layer_int (0000)
 000D18D0 0060:
-	_code_000d18d0 (0000)
+	_draw_layer (0000)
 000D1930 0060:
-	_code_000d1930 (0000)
+	_draw_initial_layer (0000)
 000D1990 0030:
-	_code_000d1990 (0000)
+	_this_is_awful (0000)
 000D19C0 00a0:
-	_code_000d19c0 (0000)
+	_do_convoluation_coords (0000)
 000D1A60 0130:
-	_code_000d1a60 (0000)
+	_draw_fade_layer (0000)
 000D1B90 0010:
 	_progress_bar_enable (0000)
 000D1BA0 0010:
 	_code_000d1ba0 (0000)
 000D1BB0 0010:
-	_code_000d1bb0 (0000)
+	_IDirect3DTexture8_Release@4 (0000)
 000D1BC0 0010:
 	_code_000d1bc0 (0000)
 000D1BD0 0020:
@@ -99,15 +99,15 @@ symbols in this file:
 000D1E90 0070:
 	_SetTextureStageStateSmart (0000)
 000D1F00 04b0:
-	_code_000d1f00 (0000)
+	_progress_bar_render (0000)
 000D23B0 00b0:
 	_code_000d23b0 (0000)
 000D2460 00d0:
-	_code_000d2460 (0000)
+	_progress_bar_create_mask_texture (0000)
 000D2530 00c0:
 	_progress_bar_eachframe (0000)
 000D25F0 0490:
-	_code_000d25f0 (0000)
+	_progress_bar_make_stuff_ready (0000)
 000D2A80 00a0:
 	_progress_bar_display (0000)
 00271088 0148:
@@ -279,27 +279,27 @@ void generate_gravy_rect(
 	struct gravy_layer const *layer,
 	struct gravy_rectangle2d *rect);
 
-static void draw_gravy_layer(
+static void draw_layer_int(
 	struct gravy_layer const *layer,
 	real_rgb_color const *color,
 	real alpha,
 	real progress);
 
-void draw_gravy_screen_layer(
+void draw_layer(
 	real distance,
 	real alpha,
 	real_rgb_color const *color,
 	real progress);
 
-static void draw_gravy_base_layer(
+static void draw_initial_layer(
 	real_rgb_color const *color,
 	real progress);
 
-void set_blur_texture_coordinates(
+void do_convoluation_coords(
 	real x,
 	real y);
 
-void draw_fullscreen_blur(
+void draw_fade_layer(
 	real x_offset,
 	real y_offset,
 	real alpha);
@@ -310,17 +310,17 @@ static void progress_bar_load_loading_texture(
 static void progress_bar_create_noise_texture(
 	void);
 
-static void progress_bar_build_glow_texture(
+static void progress_bar_create_mask_texture(
 	void);
 
-static void progress_bar_setup(
+static void progress_bar_make_stuff_ready(
 	void);
 
-static void progress_bar_alias_surface_texture(
+static void this_is_awful(
 	D3DBaseTexture *texture,
 	D3DSurface const *surface);
 
-static void progress_bar_draw(
+static void progress_bar_render(
 	real progress);
 
 void tgaLoadHeader(
@@ -1809,17 +1809,17 @@ void progress_bar_eachframe(
 		last_t= 0;
 		if (progress_bar_mode.texture0)
 		{
-			IDirect3DResource8_Release((D3DResource *)progress_bar_mode.texture0);
+			IDirect3DTexture8_Release(progress_bar_mode.texture0);
 			progress_bar_mode.texture0= NULL;
 		}
 		if (progress_bar_mode.texture1)
 		{
-			IDirect3DResource8_Release((D3DResource *)progress_bar_mode.texture1);
+			IDirect3DTexture8_Release(progress_bar_mode.texture1);
 			progress_bar_mode.texture1= NULL;
 		}
 		if (progress_bar_mode.texture0)
 		{
-			IDirect3DResource8_Release((D3DResource *)progress_bar_mode.texture0);
+			IDirect3DTexture8_Release(progress_bar_mode.texture0);
 			progress_bar_mode.texture0= NULL;
 		}
 		for (sound_index= 0; sound_index<NUMBER_OF_PROGRESS_BAR_SOUNDS; sound_index++)
@@ -1846,13 +1846,13 @@ void progress_bar_display(
 	{
 		if (!progress_bar_mode.texture0)
 		{
-			progress_bar_setup();
+			progress_bar_make_stuff_ready();
 			progress_bar_globals.initial_progress= progress;
 		}
 
 		if (progress_bar_rendering_enabled)
 		{
-			progress_bar_draw((progress - progress_bar_globals.initial_progress) /
+			progress_bar_render((progress - progress_bar_globals.initial_progress) /
 				(1.f - progress_bar_globals.initial_progress));
 		}
 	}
@@ -1882,7 +1882,7 @@ void to_screen(
 	return;
 }
 
-static void draw_gravy_layer(
+static void draw_layer_int(
 	struct gravy_layer const *layer,
 	real_rgb_color const *color,
 	real alpha,
@@ -1915,7 +1915,7 @@ static void draw_gravy_layer(
 	return;
 }
 
-void draw_gravy_screen_layer(
+void draw_layer(
 	real distance,
 	real alpha,
 	real_rgb_color const *color,
@@ -1930,12 +1930,12 @@ void draw_gravy_screen_layer(
 	layer.half_width= (real)fabs(640.0/2.0);
 	layer.half_height= (real)fabs(480.0/2.0);
 	layer.distance= distance;
-	draw_gravy_layer(&layer, color, alpha, progress);
+	draw_layer_int(&layer, color, alpha, progress);
 
 	return;
 }
 
-static void draw_gravy_base_layer(
+static void draw_initial_layer(
 	real_rgb_color const *color,
 	real progress)
 {
@@ -1948,12 +1948,12 @@ static void draw_gravy_base_layer(
 	layer.half_width= (real)fabs(640.0/2.0);
 	layer.half_height= (real)fabs(480.0/2.0);
 	layer.distance= 0.f;
-	draw_gravy_layer(&layer, color, 1.f, progress);
+	draw_layer_int(&layer, color, 1.f, progress);
 
 	return;
 }
 
-void set_blur_texture_coordinates(
+void do_convoluation_coords(
 	real x,
 	real y)
 {
@@ -1965,7 +1965,7 @@ void set_blur_texture_coordinates(
 	return;
 }
 
-void draw_fullscreen_blur(
+void draw_fade_layer(
 	real x_offset,
 	real y_offset,
 	real alpha)
@@ -1975,17 +1975,17 @@ void draw_fullscreen_blur(
 	IDirect3DDevice8_Begin(global_d3d_device, D3DPT_TRIANGLEFAN);
 	IDirect3DDevice8_SetVertexData4f(global_d3d_device, D3DVSDE_DIFFUSE,
 		254.f/255.f, 254.f/255.f, 1.f, alpha);
-	set_blur_texture_coordinates(3.f, 0.f);
+	do_convoluation_coords(3.f, 0.f);
 	top= 1.f + (1.f - y_offset)/128.f;
 	left= (-1.f - x_offset)/128.f - 1.f;
 	IDirect3DDevice8_SetVertexData4f(global_d3d_device, D3DVSDE_VERTEX, left, top, 0.5f, 1.f);
-	set_blur_texture_coordinates(637.f, 0.f);
+	do_convoluation_coords(637.f, 0.f);
 	right= 1.f + (1.f - x_offset)/128.f;
 	IDirect3DDevice8_SetVertexData4f(global_d3d_device, D3DVSDE_VERTEX, right, top, 0.5f, 1.f);
-	set_blur_texture_coordinates(637.f, 480.f);
+	do_convoluation_coords(637.f, 480.f);
 	bottom= (-1.f - y_offset)/128.f - 1.f;
 	IDirect3DDevice8_SetVertexData4f(global_d3d_device, D3DVSDE_VERTEX, right, bottom, 0.5f, 1.f);
-	set_blur_texture_coordinates(3.f, 480.f);
+	do_convoluation_coords(3.f, 480.f);
 	IDirect3DDevice8_SetVertexData4f(global_d3d_device, D3DVSDE_VERTEX, left, bottom, 0.5f, 1.f);
 	IDirect3DDevice8_End(global_d3d_device);
 
@@ -2007,7 +2007,7 @@ void generate_gravy_rect(
 	return;
 }
 
-static void progress_bar_alias_surface_texture(
+static void this_is_awful(
 	D3DBaseTexture *texture,
 	D3DSurface const *surface)
 {
@@ -2020,7 +2020,7 @@ static void progress_bar_alias_surface_texture(
 	return;
 }
 
-static void progress_bar_draw(
+static void progress_bar_render(
 	real progress)
 {
 	real ranges[NUMBER_OF_PROGRESS_BAR_SOUNDS][2]= {{0.f, 1.f}, {0.4f, 1.f}, {0.5f, 1.f}, {0.55f, 1.f}};
@@ -2086,7 +2086,7 @@ static void progress_bar_draw(
 	IDirect3DDevice8_SetVertexShader(global_d3d_device, 0);
 	IDirect3DDevice8_SetPixelShaderProgram(global_d3d_device, (D3DPIXELSHADERDEF *)&blur_shader);
 	IDirect3DDevice8_GetBackBuffer(global_d3d_device, -1, 0, &back_buffer);
-	progress_bar_alias_surface_texture(&back_buffer_texture, back_buffer);
+	this_is_awful(&back_buffer_texture, back_buffer);
 	for (stage= 0; stage<4; stage++)
 	{
 		IDirect3DDevice8_SetTextureStageState(global_d3d_device, stage, D3DTSS_MINFILTER, D3DTEXF_LINEAR);
@@ -2098,16 +2098,16 @@ static void progress_bar_draw(
 	}
 
 	if (progress<0.9f)
-		draw_fullscreen_blur(0.f, 0.f, 0.9f);
+		draw_fade_layer(0.f, 0.f, 0.9f);
 	else
-		draw_fullscreen_blur(0.f, 0.f, 0.9f - (progress - 0.9f)*9.f);
+		draw_fade_layer(0.f, 0.f, 0.9f - (progress - 0.9f)*9.f);
 	IDirect3DDevice8_SetTexture(global_d3d_device, 0, (D3DBaseTexture *)progress_bar_mode.texture0);
 	IDirect3DDevice8_SetTexture(global_d3d_device, 1, (D3DBaseTexture *)progress_bar_mode.texture1);
 	IDirect3DDevice8_SetTextureStageState(global_d3d_device, 1, D3DTSS_BORDERCOLOR, 0x05050505);
 	IDirect3DDevice8_SetTextureStageState(global_d3d_device, 1, D3DTSS_ADDRESSU, D3DTADDRESS_BORDER);
 	IDirect3DDevice8_SetTextureStageState(global_d3d_device, 1, D3DTSS_ADDRESSV, D3DTADDRESS_CLAMP);
 	IDirect3DDevice8_SetPixelShaderProgram(global_d3d_device, (D3DPIXELSHADERDEF *)&regular_shader);
-	draw_gravy_base_layer(&color, progress);
+	draw_initial_layer(&color, progress);
 	for (stage= 0; stage<4; stage++)
 		IDirect3DDevice8_SetTexture(global_d3d_device, stage, NULL);
 	IDirect3DDevice8_SetTransform(global_d3d_device, D3DTS_WORLD, &progress_bar_globals.saved_world);
@@ -2127,7 +2127,7 @@ static void progress_bar_create_noise_texture(
 	return;
 }
 
-static void progress_bar_build_glow_texture(
+static void progress_bar_create_mask_texture(
 	void)
 {
 	D3DLOCKED_RECT locked_rect;
@@ -2156,7 +2156,7 @@ static void progress_bar_build_glow_texture(
 	return;
 }
 
-static void progress_bar_setup(
+static void progress_bar_make_stuff_ready(
 	void)
 {
 	DSBUFFERDESC buffer_description;
@@ -2172,7 +2172,7 @@ static void progress_bar_setup(
 	D3DXMatrixPerspectiveLH(&progress_bar_globals.screen_transform, 0.64f, 0.48f, 1.f, 1000.f);
 	progress_bar_load_loading_texture(&progress_bar_mode.texture0);
 	progress_bar_create_noise_texture();
-	progress_bar_build_glow_texture();
+	progress_bar_create_mask_texture();
 	IDirect3DDevice8_SetRenderState(global_d3d_device, D3DRS_ALPHABLENDENABLE, TRUE);
 	IDirect3DDevice8_SetRenderState(global_d3d_device, D3DRS_SRCBLEND, D3DBLEND_SRCALPHA);
 	IDirect3DDevice8_SetRenderState(global_d3d_device, D3DRS_DESTBLEND, D3DBLEND_ONE);
@@ -2286,7 +2286,7 @@ static void progress_bar_load_loading_texture(
 	return;
 }
 
-boolean progress_bar_screen_initialized(
+boolean progress_bar_is_stuff_ready(
 	void)
 {
 	boolean result;
diff --git a/source/interface/progress_bar.h b/source/interface/progress_bar.h
index 56642a0f..d8701bc6 100644
--- a/source/interface/progress_bar.h
+++ b/source/interface/progress_bar.h
@@ -35,7 +35,7 @@ void progress_bar_enable(
 	boolean enabled);
 void progress_bar_display(
 	real progress);
-boolean progress_bar_screen_initialized(
+boolean progress_bar_is_stuff_ready(
 	void);
 
 /* ---------- globals */
