diff --git a/config/symbols.json b/config/symbols.json
index 885b3b3d..e0740aeb 100644
--- a/config/symbols.json
+++ b/config/symbols.json
@@ -7347,7 +7347,7 @@
 { "file_offset": 1780384, "flags": 32, "name": "_saved_game_file_remember_last_used_multiplayer_map" },
 { "file_offset": 1780592, "flags": 32, "name": "_saved_game_file_retrieve_last_used_multiplayer_map" },
 { "file_offset": 1780816, "flags": 32, "name": "_saved_game_file_generate_checksum" },
-{ "file_offset": 1780960, "flags": 32, "name": "_find_or_create_directory", "static": true },
+{ "file_offset": 1780960, "flags": 32, "name": "_find_and_create_directory_if_necessary", "static": true },
 { "file_offset": 1781056, "flags": 32, "name": "_enumerate_mapfile_begin", "static": true },
 { "file_offset": 1781264, "flags": 32, "name": "_enumerate_mapfile_end", "static": true },
 { "file_offset": 1781408, "flags": 32, "name": "_append_entry_to_mapfile", "static": true },
@@ -7360,8 +7360,8 @@
 { "file_offset": 1782272, "flags": 32, "name": "_build_saved_game_file_index", "static": true },
 { "file_offset": 1782336, "flags": 32, "name": "_saved_game_files_initialize" },
 { "file_offset": 1782720, "flags": 32, "name": "_saved_game_file_find_profile_index_for_directory_path" },
-{ "file_offset": 1783072, "flags": 32, "name": "_enumerate_default_playlist_profile_files", "static": true },
-{ "file_offset": 1783600, "flags": 32, "name": "_enumerate_default_player_profile_files", "static": true },
+{ "file_offset": 1783072, "flags": 32, "name": "_enumerate_default_playlist_profiles", "static": true },
+{ "file_offset": 1783600, "flags": 32, "name": "_enumerate_default_player_profiles", "static": true },
 { "file_offset": 1784112, "flags": 32, "name": "_get_nth_entry_in_mapfile", "static": true },
 { "file_offset": 1784608, "flags": 32, "name": "_set_nth_entry_in_mapfile", "static": true },
 { "file_offset": 1785088, "flags": 32, "name": "_add_new_entry_to_mapfile", "static": true },
@@ -7371,7 +7371,7 @@
 { "file_offset": 1786688, "flags": 32, "name": "_saved_game_file_open" },
 { "file_offset": 1787008, "flags": 32, "name": "_synchronize_metadata_display_name_with_profile_name" },
 { "file_offset": 1788048, "flags": 32, "name": "_saved_game_file_get_path_to_enclosing_directory" },
-{ "file_offset": 1788416, "flags": 32, "name": "_saved_game_files_enumerate_default_files" },
+{ "file_offset": 1788416, "flags": 32, "name": "_enumerate_default_profiles" },
 { "file_offset": 1788448, "flags": 32, "name": "_saved_game_files_delete_all_custom_profiles" },
 { "file_offset": 1788672, "flags": 32, "name": "_enumerate_memory_units" },
 { "file_offset": 1789664, "flags": 32, "name": "_saved_game_files_enumerate_available_to_local_player_index" },
diff --git a/source/saved games/saved_game_files.c b/source/saved games/saved_game_files.c
index 2477e104..36f17693 100644
--- a/source/saved games/saved_game_files.c	
+++ b/source/saved games/saved_game_files.c	
@@ -35,7 +35,7 @@ symbols in this file:
 001B2C50 0090:
 	_saved_game_file_generate_checksum (0000)
 001B2CE0 0060:
-	_code_001b2ce0 (0000)
+	_find_and_create_directory_if_necessary (0000)
 001B2D40 00d0:
 	_code_001b2d40 (0000)
 001B2E10 0090:
@@ -61,9 +61,9 @@ symbols in this file:
 001B33C0 0160:
 	_saved_game_file_find_profile_index_for_directory_path (0000)
 001B3520 0210:
-	_code_001b3520 (0000)
+	_enumerate_default_playlist_profiles (0000)
 001B3730 0200:
-	_code_001b3730 (0000)
+	_enumerate_default_player_profiles (0000)
 001B3930 01f0:
 	_code_001b3930 (0000)
 001B3B20 01e0:
@@ -83,7 +83,7 @@ symbols in this file:
 001B4890 0170:
 	_saved_game_file_get_path_to_enclosing_directory (0000)
 001B4A00 0020:
-	_code_001b4a00 (0000)
+	_enumerate_default_profiles (0000)
 001B4A20 00e0:
 	_saved_game_files_delete_all_custom_profiles (0000)
 001B4B00 03e0:
@@ -433,7 +433,7 @@ typedef char verify_saved_game_files_globals_size[
 
 /* ---------- prototypes */
 
-static boolean find_or_create_directory(
+static boolean find_and_create_directory_if_necessary(
 	char const *path);
 static boolean append_entry_to_mapfile(
 	struct enumerated_saved_game_file *file);
@@ -455,9 +455,9 @@ static long build_saved_game_file_index(
 	boolean valid);
 static boolean enumerate_mapfile_end(
 	word memory_unit);
-static short enumerate_default_playlist_profile_files(
+static short enumerate_default_playlist_profiles(
 	void);
-static short enumerate_default_player_profile_files(
+static short enumerate_default_player_profiles(
 	void);
 static boolean get_nth_entry_in_mapfile(
 	word memory_unit_index,
@@ -497,37 +497,37 @@ struct saved_game_files_globals saved_game_files_globals = {0};
 void saved_game_files_initialize(
 	void)
 {
-	if (!find_or_create_directory("z:\\saved"))
+	if (!find_and_create_directory_if_necessary("z:\\saved"))
 	{
 		error(_error_silent, "failed to find/create '%s' directory", "z:\\saved");
 	}
 
-	if (!find_or_create_directory("z:\\saved\\player_profiles"))
+	if (!find_and_create_directory_if_necessary("z:\\saved\\player_profiles"))
 	{
 		error(_error_silent, "failed to find/create '%s' directory", "z:\\saved\\player_profiles");
 	}
 
-	if (!find_or_create_directory("z:\\saved\\player_profiles\\default_profile"))
+	if (!find_and_create_directory_if_necessary("z:\\saved\\player_profiles\\default_profile"))
 	{
 		error(_error_silent, "failed to find/create '%s' directory", "z:\\saved\\player_profiles\\default_profile");
 	}
 
-	if (!find_or_create_directory("z:\\saved\\playlists"))
+	if (!find_and_create_directory_if_necessary("z:\\saved\\playlists"))
 	{
 		error(_error_silent, "failed to find/create '%s' directory", "z:\\saved\\playlists");
 	}
 
-	if (!find_or_create_directory("z:\\saved\\playlists\\default_playlist"))
+	if (!find_and_create_directory_if_necessary("z:\\saved\\playlists\\default_playlist"))
 	{
 		error(_error_silent, "failed to find/create '%s' directory", "z:\\saved\\playlists\\default_playlist");
 	}
 
-	if (!find_or_create_directory("z:\\saved\\recordings"))
+	if (!find_and_create_directory_if_necessary("z:\\saved\\recordings"))
 	{
 		error(_error_silent, "failed to find/create '%s' directory", "z:\\saved\\recordings");
 	}
 
-	if (!find_or_create_directory("z:\\saved\\recordings\\last_recording"))
+	if (!find_and_create_directory_if_necessary("z:\\saved\\recordings\\last_recording"))
 	{
 		error(_error_silent, "failed to find/create '%s' directory", "z:\\saved\\recordings\\last_recording");
 	}
@@ -1321,11 +1321,11 @@ static boolean set_nth_entry_in_mapfile(
 	return success;
 }
 
-short saved_game_files_enumerate_default_files(
+short enumerate_default_profiles(
 	void)
 {
-	short number_of_playlist_files = enumerate_default_playlist_profile_files();
-	short number_of_player_profile_files = enumerate_default_player_profile_files();
+	short number_of_playlist_files = enumerate_default_playlist_profiles();
+	short number_of_player_profile_files = enumerate_default_player_profiles();
 
 	return number_of_playlist_files+number_of_player_profile_files;
 }
@@ -1787,8 +1787,8 @@ void saved_game_files_delete_all_custom_profiles(
 				}
 
 				{
-					short number_of_playlist_files = enumerate_default_playlist_profile_files();
-					short number_of_player_profile_files = enumerate_default_player_profile_files();
+					short number_of_playlist_files = enumerate_default_playlist_profiles();
+					short number_of_player_profile_files = enumerate_default_player_profiles();
 					short number_of_default_files = number_of_playlist_files+number_of_player_profile_files;
 
 					number_of_enumerated_files += number_of_default_files;
@@ -1947,8 +1947,8 @@ void enumerate_memory_units(
 					}
 
 					{
-						short number_of_playlist_files = enumerate_default_playlist_profile_files();
-						short number_of_player_profile_files = enumerate_default_player_profile_files();
+						short number_of_playlist_files = enumerate_default_playlist_profiles();
+						short number_of_player_profile_files = enumerate_default_player_profiles();
 						short number_of_default_files = number_of_playlist_files+number_of_player_profile_files;
 
 						number_of_enumerated_files += number_of_default_files;
@@ -1978,7 +1978,7 @@ void enumerate_memory_units_test(
 
 /* ---------- private code */
 
-static boolean find_or_create_directory(
+static boolean find_and_create_directory_if_necessary(
 	char const *path)
 {
 	struct file_reference directory;
@@ -2208,7 +2208,7 @@ static long build_saved_game_file_index(
 	return profile_index;
 }
 
-static short enumerate_default_playlist_profile_files(
+static short enumerate_default_playlist_profiles(
 	void)
 {
 	byte block[SAVED_GAME_FILE_BLOCK_SIZE];
@@ -2286,7 +2286,7 @@ static short enumerate_default_playlist_profile_files(
 	return profile_index;
 }
 
-static short enumerate_default_player_profile_files(
+static short enumerate_default_player_profiles(
 	void)
 {
 	byte block[SAVED_GAME_FILE_BLOCK_SIZE];
diff --git a/source/saved games/saved_game_files.h b/source/saved games/saved_game_files.h
index 23f0fe5e..06909324 100644
--- a/source/saved games/saved_game_files.h	
+++ b/source/saved games/saved_game_files.h	
@@ -87,7 +87,7 @@ boolean synchronize_metadata_display_name_with_profile_name(
 boolean saved_game_file_get_path_to_enclosing_directory(
 	long profile_index,
 	char *directory_path);
-short saved_game_files_enumerate_default_files(
+short enumerate_default_profiles(
 	void);
 long create_enumerated_saved_game_file(
 	word saved_game_file_type,
