diff --git a/config/symbols.json b/config/symbols.json
index 885b3b3d..30961b42 100644
--- a/config/symbols.json
+++ b/config/symbols.json
@@ -7221,7 +7221,7 @@
 { "file_offset": 1760304, "flags": 32, "name": "_sound_cache_sound_hardware_unlock" },
 { "file_offset": 1760352, "flags": 32, "name": "_sound_cache_locked_block_proc" },
 { "file_offset": 1760416, "flags": 32, "name": "_sound_cache_delete_block_proc" },
-{ "file_offset": 1760608, "flags": 32, "name": "_sound_cache_debug_block_name" },
+{ "file_offset": 1760608, "flags": 32, "name": "_cache_block_get_sound_permutation_name" },
 { "file_offset": 1760672, "flags": 32, "name": "_sound_cache_start_loading_sound" },
 { "file_offset": 1760976, "flags": 32, "name": "_sound_cache_new" },
 { "file_offset": 1761152, "flags": 32, "name": "_sound_cache_flush" },
diff --git a/source/cache/xbox_sound_cache.c b/source/cache/xbox_sound_cache.c
index fd7b7224..642529f7 100644
--- a/source/cache/xbox_sound_cache.c
+++ b/source/cache/xbox_sound_cache.c
@@ -23,7 +23,7 @@ symbols in this file:
 001ADCA0 00c0:
 	_sound_cache_delete_block_proc (0000)
 001ADD60 0040:
-	_sound_cache_debug_block_name (0000)
+	_cache_block_get_sound_permutation_name (0000)
 001ADDA0 0130:
 	_sound_cache_start_loading_sound (0000)
 001ADED0 00b0:
@@ -468,7 +468,7 @@ void sound_cache_delete_block_proc(
 	return;
 }
 
-const char *sound_cache_debug_block_name(
+const char *cache_block_get_sound_permutation_name(
 	long block_index)
 {
 	struct xbox_cache_sound_datum *cache_sound;
@@ -541,7 +541,7 @@ static void sound_cache_start_loading_sound(
 			sound->samples.size,
 			xbox_sound_cache_globals.cache,
 			scenario_debug_to_file,
-			sound_cache_debug_block_name);
+			cache_block_get_sound_permutation_name);
 		xbox_sound_cache_globals.last_allocation_failure_time =
 			system_milliseconds();
 	}
