--- a/source/interface/player_ui.h
+++ b/source/interface/player_ui.h
@@ -19,7 +19,6 @@
 /* ---------- structures */
 
 struct player_profile;
-struct playlist_profile;
 struct game_variant;
 
 /* ---------- prototypes/PLAYER_UI.C */
@@ -44,7 +43,7 @@
 	short local_player_index);
 struct player_profile *player_ui_get_edit_player_profile(
 	void);
-struct playlist_profile *player_ui_get_edit_playlist_profile(
+struct game_variant *player_ui_get_edit_playlist_profile(
 	void);
 boolean player_ui_edit_profile_is_dirty(
 	void);
