--- a/source/interface/interface.h	2026-09-23 19:56:24.304077300 -0700
+++ b/source/interface/interface.h	2026-09-23 19:56:24.487680700 -0700
@@ -51,8 +51,6 @@
 	short interface_color_table_index,
 	short color_index,
 	real_argb_color *color);
-long interface_get_weapon_hud_index(
-	real *flashlight_power);
 void interface_set_bitmap_text_draw_mode(
 	short interface_font_index,
 	short style,
