diff --git a/source/interface/hud_unit.c b/source/interface/hud_unit.c
index cfa13729..73d9957f 100644
--- a/source/interface/hud_unit.c
+++ b/source/interface/hud_unit.c
@@ -113,12 +113,6 @@ enum hud_damage_indicator_direction
 	NUMBER_OF_HUD_DAMAGE_INDICATOR_DIRECTIONS
 };
 
-enum
-{
-	STACK_BUFFER_LENGTH = 0x80,
-	STACK_BUFFER_FILL = 0x62626262,
-};
-
 /* ---------- macros */
 
 /* ---------- structures */
@@ -672,34 +666,7 @@ update_finished:
 		}
 	}
 
-	{
-		short corrupt_index;
-		short buffer_index;
-
-		for (buffer_index = STACK_BUFFER_LENGTH - 1; buffer_index >= 0; buffer_index--)
-		{
-			if (stack_buffer[buffer_index] != STACK_BUFFER_FILL)
-				goto corrupt_stack_found;
-		}
-
-		corrupt_index = NONE;
-		goto stack_buffer_checked;
-
-corrupt_stack_found:
-		corrupt_index = buffer_index;
-
-stack_buffer_checked:
-		match_vassert(
-			"c:\\halo\\SOURCE\\interface\\hud_unit.c",
-			0x201,
-			return_eip == get_return_eip(),
-			"corrupt return address!");
-		match_vassert(
-			"c:\\halo\\SOURCE\\interface\\hud_unit.c",
-			0x201,
-			corrupt_index == NONE,
-			csprintf(temporary, "corrupt stack at %d!", corrupt_index));
-	}
+	match_assert_stack_frame("c:\\halo\\SOURCE\\interface\\hud_unit.c", 0x201);
 
 	return;
 }
@@ -1439,34 +1406,7 @@ void hud_render_unit_interface(
 		}
 	}
 
-	{
-		short corrupt_index;
-		short buffer_index;
-
-		for (buffer_index = STACK_BUFFER_LENGTH - 1; buffer_index >= 0; buffer_index--)
-		{
-			if (stack_buffer[buffer_index] != STACK_BUFFER_FILL)
-				goto corrupt_stack_found;
-		}
-
-		corrupt_index = NONE;
-		goto stack_buffer_checked;
-
-corrupt_stack_found:
-		corrupt_index = buffer_index;
-
-stack_buffer_checked:
-		match_vassert(
-			"c:\\halo\\SOURCE\\interface\\hud_unit.c",
-			0x3C9,
-			return_eip == get_return_eip(),
-			"corrupt return address!");
-		match_vassert(
-			"c:\\halo\\SOURCE\\interface\\hud_unit.c",
-			0x3C9,
-			corrupt_index == NONE,
-			csprintf(temporary, "corrupt stack at %d!", corrupt_index));
-	}
+	match_assert_stack_frame("c:\\halo\\SOURCE\\interface\\hud_unit.c", 0x3C9);
 
 	return;
 }
