diff --git a/source/interface/motion_sensor.c b/source/interface/motion_sensor.c
--- a/source/interface/motion_sensor.c
+++ b/source/interface/motion_sensor.c
@@ -309,7 +309,7 @@ static void motion_sensor_update(
 	void)
 {
 	long return_eip = get_return_eip();
-	long stack_buffer[0x80];
+	long stack_buffer[STACK_BUFFER_LENGTH];
 	long current_time;
 	short active_sensor_index;
 
@@ -471,34 +471,7 @@ static void motion_sensor_update(
 		}
 	}
 
-	{
-		short corrupt_index;
-		short buffer_index;
-
-		for (buffer_index = 0x7F; buffer_index >= 0; buffer_index--)
-		{
-			if (stack_buffer[buffer_index] != 0x62626262)
-				goto corrupt_stack_found_motion_sensor_update;
-		}
-
-		corrupt_index = NONE;
-		goto stack_buffer_checked_motion_sensor_update;
-
-corrupt_stack_found_motion_sensor_update:
-		corrupt_index = buffer_index;
-
-stack_buffer_checked_motion_sensor_update:
-		match_vassert(
-			"c:\\halo\\SOURCE\\interface\\motion_sensor.c",
-			642,
-			return_eip == get_return_eip(),
-			"corrupt return address!");
-		match_vassert(
-			"c:\\halo\\SOURCE\\interface\\motion_sensor.c",
-			642,
-			corrupt_index == NONE,
-			csprintf(temporary, "corrupt stack at %d!", corrupt_index));
-	}
+	match_assert_stack_frame("c:\\halo\\SOURCE\\interface\\motion_sensor.c", 642);
 
 	return;
 }
@@ -507,141 +480,127 @@ static void update_motion_sensor(
 	short local_player_index)
 {
 	long return_eip = get_return_eip();
-	long stack_buffer[0x80];
+	long stack_buffer[STACK_BUFFER_LENGTH];
 	struct motion_sensor_player *player;
 	short active_sensor_index;
-	long player_index;
 	long unit_index;
 	struct motion_sensor_datum *sensor;
-	real_point3d camera_position;
-	short blip_index;
-	real_point2d custom_positions[MAXIMUM_MOTION_SENSOR_BLIPS];
-	long out_of_range_count = 0;
 
 	csmemset(stack_buffer, 0x62, sizeof(stack_buffer));
 	player = get_motion_sensor_data(local_player_index);
 	game_time_get();
-	if (!motion_sensor_globals->update)
-		goto update_motion_sensor_done;
-
-	active_sensor_index = motion_sensor_globals->active_sensor_index;
-	player_index = local_player_get_player_index(local_player_index);
-	unit_index = player_index == NONE
-		? NONE
-		: player_get(local_player_get_player_index(local_player_index))->unit_index;
-	sensor = &player->sensor_data[active_sensor_index];
-	if (unit_index == NONE)
-		goto update_motion_sensor_done;
-
-	unit_get_camera_position(unit_index, &camera_position);
-	sensor->reference_point.x = camera_position.x;
-	sensor->reference_point.y = camera_position.y;
-
-	for (blip_index = 0;
-		blip_index < MAXIMUM_MOTION_SENSOR_BLIPS;
-		blip_index++)
+	if (motion_sensor_globals->update)
 	{
-		long object_index = player->unit_indices[blip_index];
-
-		if (unit_try_and_get(object_index))
+		active_sensor_index = motion_sensor_globals->active_sensor_index;
+		unit_index = local_player_get_player_index(local_player_index) == NONE
+			? NONE
+			: player_get(local_player_get_player_index(local_player_index))->unit_index;
+		sensor = &player->sensor_data[active_sensor_index];
+		if (unit_index != NONE)
 		{
-			real_point3d object_position;
-			real object_radius;
-			real_point2d displacement;
-			boolean draw = should_draw_object(object_index);
-
-			object_get_bounding_sphere(
-				object_index,
-				&object_position,
-				&object_radius);
-			displacement.x = object_position.x - sensor->reference_point.x;
-			displacement.y = object_position.y - sensor->reference_point.y;
-
-			if (!draw ||
-				magnitude_squared2d((real_vector2d const *)&displacement) >
-					hud_globals->defaults.motion_sensor_range *
-						hud_globals->defaults.motion_sensor_range)
 			{
-				sensor->blips[blip_index].type = _blip_type_none;
-				player->unit_indices[blip_index] = NONE;
+				real_point3d reference_point;
+
+				unit_get_camera_position(unit_index, &reference_point);
+				sensor->reference_point.x = reference_point.x;
+				sensor->reference_point.y = reference_point.y;
 			}
-			else
+
 			{
-				tiny_point2d_set(
-					&sensor->blips[blip_index].position,
-					&displacement);
-			}
-		}
-	}
+				short blip_index;
+				struct motion_sensor_blip *blips = sensor->blips;
 
-	sensor->yaw =
-		player_control_get_facing_angles(local_player_index)->yaw +
-		1.5707964f;
-	player_index = local_player_get_player_index(local_player_index);
-	sensor->custom_blip_count =
-		(byte)game_engine_player_get_custom_motion_sensor_positions(
-			player_index,
-			custom_positions,
-			(byte *)sensor->custom_blip_goal_indices,
-			MAXIMUM_MOTION_SENSOR_BLIPS);
-	unit_get_camera_position(unit_index, &camera_position);
-
-	for (blip_index = 0;
-		blip_index < sensor->custom_blip_count;
-		blip_index++)
-	{
-		real_point2d *position = &custom_positions[blip_index];
-		real_point2d displacement;
-
-		displacement.x = position->x - camera_position.x;
-		displacement.y = position->y - camera_position.y;
-		*position = displacement;
-		if (magnitude_squared2d((real_vector2d const *)&displacement) >
-				hud_globals->defaults.motion_sensor_range *
-					hud_globals->defaults.motion_sensor_range)
-		{
-			out_of_range_count++;
-		}
-		else
-		{
-			tiny_point2d_set(
-				&sensor->custom_blips[blip_index - out_of_range_count],
-				position);
-		}
-	}
+				for (blip_index = 0;
+					blip_index < MAXIMUM_MOTION_SENSOR_BLIPS;
+					blip_index++)
+				{
+					long object_index = player->unit_indices[blip_index];
 
-	sensor->custom_blip_count -= (byte)out_of_range_count;
+					if (unit_try_and_get(object_index))
+					{
+						real_point3d unit_center;
+						real_point3d player_center;
+						real_point2d point;
+						real dummy;
+						boolean draw = should_draw_object(object_index);
+
+						object_get_bounding_sphere(
+							object_index,
+							&unit_center,
+							&dummy);
+						point.x = unit_center.x - sensor->reference_point.x;
+						point.y = unit_center.y - sensor->reference_point.y;
+						player_center.x = sensor->reference_point.x;
+						player_center.y = sensor->reference_point.y;
+						player_center.z = unit_center.z;
+
+						if (draw &&
+							distance_squared3d(&player_center, &unit_center) <=
+								hud_globals->defaults.motion_sensor_range *
+									hud_globals->defaults.motion_sensor_range)
+						{
+							tiny_point2d_set(
+								&blips[blip_index].position,
+								&point);
+						}
+						else
+						{
+							blips[blip_index].type = _blip_type_none;
+							player->unit_indices[blip_index] = NONE;
+						}
+					}
+				}
+			}
 
-update_motion_sensor_done:
-	{
-		short corrupt_index;
-		short buffer_index;
+			sensor->yaw =
+				player_control_get_facing_angles(local_player_index)->yaw +
+				1.5707964f;
+			{
+				real_point3d reference_point;
+				real_point2d customs[MAXIMUM_MOTION_SENSOR_BLIPS];
+				long custom_index;
+				long out_of_range_count = 0;
+
+				sensor->custom_blip_count =
+					(byte)game_engine_player_get_custom_motion_sensor_positions(
+						local_player_get_player_index(local_player_index),
+						customs,
+						(byte *)sensor->custom_blip_goal_indices,
+						MAXIMUM_MOTION_SENSOR_BLIPS);
+				unit_get_camera_position(unit_index, &reference_point);
+
+				for (custom_index = 0;
+					custom_index < sensor->custom_blip_count;
+					custom_index++)
+				{
+					real_point3d custom_temp;
+
+					custom_temp.x = customs[custom_index].x;
+					custom_temp.y = customs[custom_index].y;
+					custom_temp.z = reference_point.z;
+					customs[custom_index].x -= reference_point.x;
+					customs[custom_index].y -= reference_point.y;
+					if (distance_squared3d(&reference_point, &custom_temp) <=
+							hud_globals->defaults.motion_sensor_range *
+								hud_globals->defaults.motion_sensor_range)
+					{
+						tiny_point2d_set(
+							&sensor->custom_blips[custom_index - out_of_range_count],
+							&customs[custom_index]);
+					}
+					else
+					{
+						out_of_range_count++;
+					}
+				}
 
-		for (buffer_index = 0x7F; buffer_index >= 0; buffer_index--)
-		{
-			if (stack_buffer[buffer_index] != 0x62626262)
-				goto corrupt_stack_found_update_motion_sensor;
+				sensor->custom_blip_count -= (byte)out_of_range_count;
+			}
 		}
-
-		corrupt_index = NONE;
-		goto stack_buffer_checked_update_motion_sensor;
-
-corrupt_stack_found_update_motion_sensor:
-		corrupt_index = buffer_index;
-
-stack_buffer_checked_update_motion_sensor:
-		match_vassert(
-			"c:\\halo\\SOURCE\\interface\\motion_sensor.c",
-			756,
-			return_eip == get_return_eip(),
-			"corrupt return address!");
-		match_vassert(
-			"c:\\halo\\SOURCE\\interface\\motion_sensor.c",
-			756,
-			corrupt_index == NONE,
-			csprintf(temporary, "corrupt stack at %d!", corrupt_index));
 	}
 
+	match_assert_stack_frame("c:\\halo\\SOURCE\\interface\\motion_sensor.c", 756);
+
 	return;
 }
 
@@ -651,7 +610,7 @@ static void render_motion_sensor(
 	point2d const *reference)
 {
 	long return_eip = get_return_eip();
-	long stack_buffer[0x80];
+	long stack_buffer[STACK_BUFFER_LENGTH];
 	director_perspective perspective;
 	struct motion_sensor_player *player;
 	real relative_scale;
@@ -743,34 +702,7 @@ static void render_motion_sensor(
 	rasterizer_hud_motion_sensor_blip_end(&center_point, sweep_theta);
 
 render_motion_sensor_done:
-	{
-		short corrupt_index;
-		short buffer_index;
-
-		for (buffer_index = 0x7F; buffer_index >= 0; buffer_index--)
-		{
-			if (stack_buffer[buffer_index] != 0x62626262)
-				goto corrupt_stack_found_render_motion_sensor;
-		}
-
-		corrupt_index = NONE;
-		goto stack_buffer_checked_render_motion_sensor;
-
-corrupt_stack_found_render_motion_sensor:
-		corrupt_index = buffer_index;
-
-stack_buffer_checked_render_motion_sensor:
-		match_vassert(
-			"c:\\halo\\SOURCE\\interface\\motion_sensor.c",
-			830,
-			return_eip == get_return_eip(),
-			"corrupt return address!");
-		match_vassert(
-			"c:\\halo\\SOURCE\\interface\\motion_sensor.c",
-			830,
-			corrupt_index == NONE,
-			csprintf(temporary, "corrupt stack at %d!", corrupt_index));
-	}
+	match_assert_stack_frame("c:\\halo\\SOURCE\\interface\\motion_sensor.c", 830);
 
 	return;
 }
