diff --git a/source/game/game_engine_king.c b/source/game/game_engine_king.c
index 526c8e0212f9609f174b0b97489293aa6515aa84..2c4cbe23f9e4bb6a0397b7b659f92660f7a71b61 100644
--- a/source/game/game_engine_king.c
+++ b/source/game/game_engine_king.c
@@ -241,55 +241,55 @@ static void king_calculate_hill_state(
 extern real_rgb_color global_default_animation_colors[4];
 extern real global_default_animation_values[4];
 
-struct king_globals king_globals = { 0 };
+static struct king_globals king_globals = { 0 };
 static short king_engine_hill_count = 0;
 static short king_engine_hills[MAXIMUM_HILLS] = { 0 };
 
 /* ---------- public code */
 
-void king_engine_dispose(
+static void king_engine_dispose(
 	void)
 {
 	return;
 }
 
-void king_engine_dispose_from_old_map(
+static void king_engine_dispose_from_old_map(
 	void)
 {
 	return;
 }
 
-void king_engine_game_ending(
+static void king_engine_game_ending(
 	void)
 {
 	return;
 }
 
-void king_engine_statistics_append(
+static void king_engine_statistics_append(
 	long statistic)
 {
 	return;
 }
 
-void king_engine_handle_client_message(
+static void king_engine_handle_client_message(
 	void *message)
 {
 	return;
 }
 
-void king_engine_handle_server_message(
+static void king_engine_handle_server_message(
 	void *message)
 {
 	return;
 }
 
-void king_engine_pregame_post_rasterize(
+static void king_engine_pregame_post_rasterize(
 	void)
 {
 	return;
 }
 
-void king_engine_post_rasterize(
+static void king_engine_post_rasterize(
 	void)
 {
 	struct game_globals *game_globals;
@@ -374,7 +374,7 @@ void king_engine_post_rasterize(
 	return;
 }
 
-void king_engine_player_damaged_player(
+static void king_engine_player_damaged_player(
 	long damaging_player_index,
 	long dead_player_index,
 	boolean damage_type)
@@ -382,7 +382,7 @@ void king_engine_player_damaged_player(
 	return;
 }
 
-void king_engine_player_killed_player(
+static void king_engine_player_killed_player(
 	long killing_player_index,
 	long killing_object_index,
 	long dead_player_index,
@@ -391,13 +391,13 @@ void king_engine_player_killed_player(
 	return;
 }
 
-void king_engine_prespawn_player_update(
+static void king_engine_prespawn_player_update(
 	long player_index)
 {
 	return;
 }
 
-void king_engine_player_added(
+static void king_engine_player_added(
 	long player_index)
 {
 	player_get(player_index);
@@ -405,7 +405,7 @@ void king_engine_player_added(
 	return;
 }
 
-void king_engine_game_starting(
+static void king_engine_game_starting(
 	void)
 {
 	game_engine_play_multiplayer_sound(
@@ -416,7 +416,7 @@ void king_engine_game_starting(
 	return;
 }
 
-wchar_t *king_get_score_string(
+static wchar_t *king_get_score_string(
 	long player_index,
 	wchar_t *buffer)
 {
@@ -430,7 +430,7 @@ wchar_t *king_get_score_string(
 	return buffer;
 }
 
-wchar_t *king_get_team_score_string(
+static wchar_t *king_get_team_score_string(
 	long team_index,
 	wchar_t *buffer)
 {
@@ -442,7 +442,7 @@ wchar_t *king_get_team_score_string(
 	return buffer;
 }
 
-boolean king_engine_initialize_for_new_map(
+static boolean king_engine_initialize_for_new_map(
 	void)
 {
 	struct scenario *scenario = global_scenario_get();
@@ -490,7 +490,7 @@ boolean king_engine_initialize_for_new_map(
 	return TRUE;
 }
 
-void king_engine_player_update(
+static void king_engine_player_update(
 	long player_index)
 {
 	struct player_datum *player = player_get(player_index);
@@ -558,7 +558,7 @@ void king_engine_player_update(
 	return;
 }
 
-boolean king_engine_display_score(
+static boolean king_engine_display_score(
 	long player_index,
 	long message,
 	long message_player_index,
@@ -639,7 +639,7 @@ boolean king_engine_display_score(
 	return result;
 }
 
-long king_get_score(
+static long king_get_score(
 	long player_index,
 	enum get_score_type score_type)
 {
@@ -779,7 +779,7 @@ void render_dynamic_quad(
 	return;
 }
 
-wchar_t *king_get_score_header_string(
+static wchar_t *king_get_score_header_string(
 	wchar_t *buffer)
 {
 	long string_list_index = tag_loaded('ustr', "ui\\multiplayer_game_text");
@@ -800,7 +800,7 @@ wchar_t *king_get_score_header_string(
 	return buffer;
 }
 
-boolean king_engine_goal_matches_player(
+static boolean king_engine_goal_matches_player(
 	long player_index,
 	long goal_index)
 {
@@ -809,7 +809,7 @@ boolean king_engine_goal_matches_player(
 	return matches;
 }
 
-void king_engine_update(
+static void king_engine_update(
 	void)
 {
 	if (game_engine_can_score() &&
diff --git a/source/game/game_engine_king.h b/source/game/game_engine_king.h
index 25f673af7a309b83212c2d46f94dc244aaa18f3f..835a3c9c2882192a82506ebcd157107b8879c9f9 100644
--- a/source/game/game_engine_king.h
+++ b/source/game/game_engine_king.h
@@ -63,85 +63,6 @@ typedef char verify_king_globals_hill_id_offset[
 
 /* ---------- prototypes/GAME_ENGINE_KING.C */
 
-void king_engine_dispose(
-	void);
-
-boolean king_engine_initialize_for_new_map(
-	void);
-
-void king_engine_dispose_from_old_map(
-	void);
-
-void king_engine_player_added(
-	long player_index);
-
-void king_engine_game_ending(
-	void);
-
-void king_engine_game_starting(
-	void);
-
-void king_engine_statistics_append(
-	long statistic);
-
-void king_engine_handle_client_message(
-	void *message);
-
-void king_engine_handle_server_message(
-	void *message);
-
-void king_engine_pregame_post_rasterize(
-	void);
-
-void king_engine_post_rasterize(
-	void);
-
-void king_engine_player_update(
-	long player_index);
-
-void king_engine_player_damaged_player(
-	long damaging_player_index,
-	long dead_player_index,
-	boolean damage_type);
-
-void king_engine_player_killed_player(
-	long killing_player_index,
-	long killing_object_index,
-	long dead_player_index,
-	boolean friendly_fire);
-
-boolean king_engine_display_score(
-	long player_index,
-	long message,
-	long message_player_index,
-	wchar_t *buffer,
-	long buffer_size);
-
-void king_engine_prespawn_player_update(
-	long player_index);
-
-long king_get_score(
-	long player_index,
-	enum get_score_type score_type);
-
-wchar_t *king_get_score_string(
-	long player_index,
-	wchar_t *buffer);
-
-wchar_t *king_get_score_header_string(
-	wchar_t *buffer);
-
-wchar_t *king_get_team_score_string(
-	long team_index,
-	wchar_t *buffer);
-
-boolean king_engine_goal_matches_player(
-	long player_index,
-	long goal_index);
-
-void king_engine_update(
-	void);
-
 void render_dynamic_quad_initialize(
 	void);
 
@@ -156,6 +77,5 @@ void render_dynamic_quad(
 /* ---------- globals */
 
 extern struct game_engine king_engine;
-extern struct king_globals king_globals;
 
 #endif // __GAME_ENGINE_KING_H
