diff --git a/source/hs/hs_runtime.c b/source/hs/hs_runtime.c
index c85fc348..402b21a9 100644
--- a/source/hs/hs_runtime.c
+++ b/source/hs/hs_runtime.c
@@ -396,7 +396,6 @@ typedef void (*hs_inspection_procedure)(
 struct hs_runtime_globals
 {
 	boolean initialized;
-	byte pad;
 	short executing_thread_index;
 };
 
@@ -547,8 +546,8 @@ static void hs_global_reconcile_write(
 extern struct data_array *hs_global_data;
 extern struct data_array *hs_thread_data;
 extern struct data_array *hs_syntax_data;
-extern short hs_external_global_count;
-extern short hs_type_sizes[NUMBER_OF_HS_TYPES];
+extern short const hs_external_global_count;
+extern short const hs_type_sizes[NUMBER_OF_HS_TYPES];
 extern boolean debug_scripting;
 extern unsigned long hs_debug_data[];
 static hs_inspection_procedure hs_type_inspectors[NUMBER_OF_HS_TYPES] =
@@ -693,45 +692,45 @@ hs_typecasting_procedure typecasting_procedures[NUMBER_OF_HS_TYPES][NUMBER_OF_HS
 static struct hs_runtime_globals hs_runtime_globals;
 
 /* per-type fallbacks used when an external global has no backing address */
-extern boolean _hs_type_boolean_default;
-extern real _hs_type_real_default;
-extern short _hs_type_short_integer_default;
-extern long _hs_type_long_integer_default;
+extern boolean const _hs_type_boolean_default;
+extern real const _hs_type_real_default;
+extern short const _hs_type_short_integer_default;
+extern long const _hs_type_long_integer_default;
 extern char const *_hs_type_string_default;
-extern short _hs_type_script_default;
-extern short _hs_type_trigger_volume_default;
-extern short _hs_type_cutscene_flag_default;
-extern short _hs_type_cutscene_camera_point_default;
-extern short _hs_type_cutscene_title_default;
-extern short _hs_type_cutscene_recording_default;
-extern short _hs_type_device_group_default;
-extern long _hs_type_ai_default;
-extern short _hs_type_ai_command_list_default;
-extern short _hs_type_starting_profile_default;
-extern short _hs_type_conversation_default;
-extern short _hs_type_navpoint_default;
-extern short _hs_type_hud_message_default;
-extern long _hs_type_object_list_default;
-extern long _hs_type_sound_default;
-extern long _hs_type_effect_default;
-extern long _hs_type_damage_default;
-extern long _hs_type_looping_sound_default;
-extern long _hs_type_animation_graph_default;
-extern long _hs_type_actor_variant_default;
-extern long _hs_type_damage_effect_default;
-extern long _hs_type_object_definition_default;
-extern short _hs_type_enum_game_difficulty_default;
-extern short _hs_type_enum_team_default;
-extern short _hs_type_enum_ai_default_state_default;
-extern short _hs_type_enum_actor_type_default;
-extern short _hs_type_enum_hud_corner_default;
-extern long _hs_type_object_default;
-extern long _hs_type_unit_default;
-extern long _hs_type_vehicle_default;
-extern long _hs_type_weapon_default;
-extern long _hs_type_device_default;
-extern long _hs_type_scenery_default;
-extern short _hs_type_object_name_default;
+extern short const _hs_type_script_default;
+extern short const _hs_type_trigger_volume_default;
+extern short const _hs_type_cutscene_flag_default;
+extern short const _hs_type_cutscene_camera_point_default;
+extern short const _hs_type_cutscene_title_default;
+extern short const _hs_type_cutscene_recording_default;
+extern short const _hs_type_device_group_default;
+extern long const _hs_type_ai_default;
+extern short const _hs_type_ai_command_list_default;
+extern short const _hs_type_starting_profile_default;
+extern short const _hs_type_conversation_default;
+extern short const _hs_type_navpoint_default;
+extern short const _hs_type_hud_message_default;
+extern long const _hs_type_object_list_default;
+extern long const _hs_type_sound_default;
+extern long const _hs_type_effect_default;
+extern long const _hs_type_damage_default;
+extern long const _hs_type_looping_sound_default;
+extern long const _hs_type_animation_graph_default;
+extern long const _hs_type_actor_variant_default;
+extern long const _hs_type_damage_effect_default;
+extern long const _hs_type_object_definition_default;
+extern short const _hs_type_enum_game_difficulty_default;
+extern short const _hs_type_enum_team_default;
+extern short const _hs_type_enum_ai_default_state_default;
+extern short const _hs_type_enum_actor_type_default;
+extern short const _hs_type_enum_hud_corner_default;
+extern long const _hs_type_object_default;
+extern long const _hs_type_unit_default;
+extern long const _hs_type_vehicle_default;
+extern long const _hs_type_weapon_default;
+extern long const _hs_type_device_default;
+extern long const _hs_type_scenery_default;
+extern short const _hs_type_object_name_default;
 
 /* ---------- public code */
 
@@ -1519,7 +1518,7 @@ static void hs_inspect_enum(
 	union hs_conversion_result value,
 	char *result)
 {
-	struct hs_enum_definition *enum_definition;
+	struct hs_enum_definition const *enum_definition;
 
 	enum_definition = &hs_enum_table[type-_hs_type_enum_game_difficulty];
 	match_assert("c:\\halo\\source\\hs\\hs_library_internal_runtime.h", 0x27b,
