--- a/source/hs/hs_runtime.c	2026-09-26 10:54:11.838267600 -0700
+++ b/source/hs/hs_runtime.c	2026-09-26 10:54:11.797295600 -0700
@@ -307,6 +307,8 @@
 
 enum
 {
+	MAXIMUM_NUMBER_OF_HS_THREADS = 0x100,
+	MAXIMUM_NUMBER_OF_HS_GLOBALS = 0x400,
 	HS_THREAD_STACK_SIZE = 0x200
 };
 
@@ -740,12 +742,12 @@
 	short global_index;
 	long index;
 
-	hs_thread_data = game_state_data_new("hs thread", 0x100, 0x218);
-	hs_global_data = game_state_data_new("hs globals", 0x400, 8);
+	hs_thread_data = game_state_data_new("hs thread", MAXIMUM_NUMBER_OF_HS_THREADS, sizeof(struct hs_thread_datum));
+	hs_global_data = game_state_data_new("hs globals", MAXIMUM_NUMBER_OF_HS_GLOBALS, sizeof(struct hs_global_datum));
 	if (hs_thread_data && hs_global_data)
 	{
 		match_vassert("c:\\halo\\SOURCE\\hs\\hs_runtime.c", 0xa9,
-			hs_external_global_count*2<0x400,
+			hs_external_global_count*2<MAXIMUM_NUMBER_OF_HS_GLOBALS,
 			"raise MAXIMUM_NUMBER_OF_HS_GLOBALS.");
 		data_make_valid(hs_global_data);
 		for (global_index = 0;
@@ -1519,15 +1521,15 @@
 	char *result)
 {
 	struct hs_enum_definition const *enum_definition;
+	short enum_value = value.short_integer;
 
 	enum_definition = &hs_enum_table[type-_hs_type_enum_game_difficulty];
 	match_assert("c:\\halo\\source\\hs\\hs_library_internal_runtime.h", 0x27b,
 		HS_TYPE_IS_ENUM(type));
-	match_vassert("c:\\halo\\source\\hs\\hs_library_internal_runtime.h", 0x27c,
-		value.short_integer>=0 && value.short_integer<enum_definition->count,
-		"enum_value>=0 && enum_value<enum_definition->count");
+	match_assert("c:\\halo\\source\\hs\\hs_library_internal_runtime.h", 0x27c,
+		enum_value>=0 && enum_value<enum_definition->count);
 
-	sprintf(result, "%s", enum_definition->values[value.short_integer]);
+	sprintf(result, "%s", enum_definition->values[enum_value]);
 
 	return;
 }
@@ -1674,8 +1676,6 @@
 	short actual_type,
 	short desired_type)
 {
-	short object_type;
-
 	match_assert("c:\\halo\\SOURCE\\hs\\hs_runtime.c", 0x5a4,
 		actual_type==_hs_passthrough || hs_type_valid(actual_type));
 	match_assert("c:\\halo\\SOURCE\\hs\\hs_runtime.c", 0x5a5,
@@ -1688,19 +1688,25 @@
 
 	if (HS_TYPE_IS_OBJECT(desired_type))
 	{
-		object_type = desired_type - _hs_type_object;
+		short object_type = desired_type - _hs_type_object;
+		short actual_object_type;
+
 		if (HS_TYPE_IS_OBJECT(actual_type))
 		{
-			return hs_object_type_can_cast(
-				actual_type-_hs_type_object,
-				object_type);
+			actual_object_type = actual_type-_hs_type_object;
 		}
-		else if (!HS_TYPE_IS_OBJECT_NAME(actual_type))
+		else if (HS_TYPE_IS_OBJECT_NAME(actual_type))
+		{
+			actual_object_type = actual_type-_hs_type_object_name;
+		}
+		else
 		{
 			return FALSE;
 		}
 
-		goto cast_object_type;
+		return hs_object_type_can_cast(
+			actual_object_type,
+			object_type);
 	}
 	else if (HS_TYPE_IS_OBJECT_NAME(desired_type))
 	{
@@ -1709,12 +1715,9 @@
 			return FALSE;
 		}
 
-		object_type = desired_type - _hs_type_object_name;
-
-cast_object_type:
 		return hs_object_type_can_cast(
 			actual_type-_hs_type_object_name,
-			object_type);
+			desired_type-_hs_type_object_name);
 	}
 	else
 	{
