--- a/source/ai/ai_communication.c
+++ b/source/ai/ai_communication.c
@@ -6597,11 +6597,11 @@
 				}
 
 				{
-					boolean valid = TRUE;
+					boolean protagonist_invalid = FALSE;
 
 					if (protagonist_unit_index == NONE)
 					{
-						valid = FALSE;
+						protagonist_invalid = TRUE;
 					}
 					else
 					{
@@ -6609,11 +6609,11 @@
 
 						if (TEST_FLAG(protagonist_unit->object.damage_flags, _object_dead_bit))
 						{
-							valid = FALSE;
+							protagonist_invalid = TRUE;
 						}
 						else if (protagonist_unit->object.type == _object_type_vehicle)
 						{
-							valid = FALSE;
+							protagonist_invalid = TRUE;
 						}
 						else if (protagonist_unit->unit.player_index != NONE &&
 							protagonist_unit->unit.actor_index == NONE)
@@ -6625,7 +6625,7 @@
 							}
 							else
 							{
-								valid = FALSE;
+								protagonist_invalid = TRUE;
 							}
 						}
 					}
@@ -6634,16 +6634,16 @@
 					{
 						if (protagonist_actor->state.mode == _actor_mode_braindead)
 						{
-							valid = FALSE;
+							protagonist_invalid = TRUE;
 						}
 						else if (protagonist_actor->state.action == _actor_action_obey &&
 							!protagonist_actor->state.action_data.obey.allow_communication)
 						{
-							valid = FALSE;
+							protagonist_invalid = TRUE;
 						}
 					}
 
-					if (!valid)
+					if (protagonist_invalid)
 					{
 						if (ai_debug.print_communication)
 						{
