--- a/source/hs/hs.h
+++ b/source/hs/hs.h
@@ -143,7 +143,13 @@
 	short parameter_types[1];
 };
 
-struct hs_external_global_definition;
+struct hs_external_global_definition
+{
+	char const *name;
+	short type;
+	short pad;
+	void *address;
+};
 
 /* ---------- prototypes/HS.C */
 
--- a/source/hs/hs.c
+++ b/source/hs/hs.c
@@ -3333,12 +3333,6 @@
 	long value2;
 };
 
-struct hs_external_global_definition
-{
-	char const *name;
-	short type;
-};
-
 struct hud_globals_definition
 {
 	byte reserved_000[0x160];
--- a/source/hs/hs_globals_external.c
+++ b/source/hs/hs_globals_external.c
@@ -908,14 +908,6 @@
 
 /* ---------- structures */
 
-struct hs_external_global_definition
-{
-	char const *name;
-	short type;
-	short pad;
-	void *address;
-};
-
 typedef char verify_hs_external_global_definition_size[
 	sizeof(struct hs_external_global_definition) == 0xC ? 1 : -1];
 
--- a/source/hs/hs_runtime.c
+++ b/source/hs/hs_runtime.c
@@ -417,14 +417,6 @@
 	union hs_conversion_result value;
 };
 
-struct hs_external_global_definition
-{
-	char const *name;
-	short type;
-	short unused;
-	void *address;
-};
-
 struct hs_thread_datum
 {
 	short identifier;
