diff --git a/source/interface/ui_widget.c b/source/interface/ui_widget.c
index c7c08f68..1bf3c20c 100644
--- a/source/interface/ui_widget.c
+++ b/source/interface/ui_widget.c
@@ -4483,7 +4483,7 @@ void display_scenario_help(
 			match_vassert(
 				"c:\\halo\\SOURCE\\interface\\ui_widget.c",
 				2438,
-				text_box,
+				text_box && text_box->type == _ui_widget_type_text_box,
 				"expected text box widget in player help screen");
 			text_box->parameters.text_box.string_list_index = string_index;
 		}
