--- a/source/render/render_debug.c
+++ b/source/render/render_debug.c
@@ -9,15 +9,15 @@
 001782E0 0090:
 	_render_debug_polygon (0000)
 00178370 00d0:
-	_code_00178370 (0000)
+	_render_debug_add_cache_string (0000)
 00178440 0070:
-	_code_00178440 (0000)
+	_build_circle_points (0000)
 001784B0 00a0:
-	_code_001784b0 (0000)
+	_build_height_matrix (0000)
 00178550 01c0:
-	_code_00178550 (0000)
+	_build_pill_points (0000)
 00178710 0290:
-	_code_00178710 (0000)
+	_render_debug_add_cache_entry (0000)
 001789A0 0120:
 	_render_debug_point (0000)
 00178AC0 00b0:
@@ -45,17 +45,17 @@
 00179500 0130:
 	_render_debug_string_at_point (0000)
 00179630 0220:
-	_code_00179630 (0000)
+	_render_debug_camera (0000)
 00179850 0110:
-	_code_00179850 (0000)
+	_render_debug_player (0000)
 00179960 0080:
-	_code_00179960 (0000)
+	_render_debug_structure (0000)
 001799E0 01e0:
-	_code_001799e0 (0000)
+	_render_debug_bsp (0000)
 00179BC0 0070:
-	_code_00179bc0 (0000)
+	_render_debug_input (0000)
 00179C30 01a0:
-	_code_00179c30 (0000)
+	_render_debug_structure_decals (0000)
 00179DD0 00d0:
 	_render_debug_point2d (0000)
 00179EA0 0120:
@@ -127,13 +127,14 @@
 0029FE8C 001f:
 	??_C@_0BP@NHMLPPIA@can?8t?5add?5box2d?5to?5debug?5cache?$AA@ (0000)
 004B8C40 740c:
-	_bss_004b8c40 (0000)
+	_render_debug_globals (0000)
 */
 
 /* ---------- headers */
 
 #include "cseries/cseries.h"
 #include "cseries/errors.h"
+#include "ai/ai_debug.h"
 #include "ai/ai_profile.h"
 #include "ai/path.h"
 #include "cache/sound_cache.h"
@@ -148,6 +149,7 @@
 #include "math/real_math.h"
 #include "physics/bsp3d.h"
 #include "physics/collision_bsp_definitions.h"
+#include "physics/collision_debug.h"
 #include "physics/collision_usage.h"
 #include "physics/collisions.h"
 #include "rasterizer/rasterizer.h"
@@ -1955,10 +1957,11 @@
 
 	for (index = 0; index+1<NUMBER_OF_RENDER_DEBUG_CIRCLE_POINTS; index++)
 	{
-		set_real_point2d(
-			&points[index+1],
-			cosine_of_angle*points[index].x - sine_of_angle*points[index].y,
-			sine_of_angle*points[index].x + cosine_of_angle*points[index].y);
+		rotate_vector2d(
+			(real_vector2d const *)&points[index],
+			sine_of_angle,
+			cosine_of_angle,
+			(real_vector2d *)&points[index+1]);
 	}
 
 	points[NUMBER_OF_RENDER_DEBUG_CIRCLE_POINTS] = points[0];
