diff --git a/source/interface/virtual_keyboard.c b/source/interface/virtual_keyboard.c
index 3d7669ee..9f0c7c79 100644
--- a/source/interface/virtual_keyboard.c
+++ b/source/interface/virtual_keyboard.c
@@ -1101,8 +1101,7 @@ static boolean virtual_keyboard_select(
 					virtual_keyboard_globals.cursor + 1,
 					virtual_keyboard_globals.cursor,
 					buffer_size - ((byte *)virtual_keyboard_globals.cursor - (byte *)virtual_keyboard_globals.text_buffer) - sizeof(wchar_t));
-				*virtual_keyboard_globals.cursor++ = virtual_keyboard_get_character(
-					virtual_keyboard_key_layout[virtual_keyboard_globals.row][virtual_keyboard_globals.column]);
+				*virtual_keyboard_globals.cursor++ = virtual_keyboard_get_current_character();
 				if (ustrcmp(virtual_keyboard_globals.text_buffer, L".fortune") == 0)
 				{
 					unsigned long fortune_index = system_milliseconds() % NUMBER_OF_VIRTUAL_KEYBOARD_FORTUNES;
