diff --git a/source/cache/cache_files_windows.c b/scratch/w/cache_files_windows/cache_files_windows.c
index 4c8eadd2..dc517c64 100644
--- a/source/cache/cache_files_windows.c
+++ b/scratch/w/cache_files_windows/cache_files_windows.c
@@ -992,8 +992,6 @@ static void cache_files_open_cache_files(
 						811,
 						FALSE,
 						csprintf(temporary, "setup for new cache file failed (#%d)", GetLastError()));
-					CloseHandle(file);
-					file = INVALID_HANDLE_VALUE;
 				}
 			}
 		}
@@ -1006,6 +1004,12 @@ static void cache_files_open_cache_files(
 				csprintf(temporary, "couldn't open or create new cache file (#%d)", GetLastError()));
 		}
 
+		if (!valid && file != INVALID_HANDLE_VALUE)
+		{
+			CloseHandle(file);
+			file = INVALID_HANDLE_VALUE;
+		}
+
 		map_file->file = file;
 		if (valid)
 		{
